lib/rest-adapter.js

```diff
@@ -143,17 +143,36 @@ RestAdapter.prototype.invoke = function(method, ctorArgs, args, callback) {
   });
 };
 
-RestAdapter.prototype.getRestMethodByName = function(name) {
-  var classes = this.getClasses();
-  for (var i = 0; i < classes.length; i++) {
-    var restClass = classes[i];
-    for (var j = 0; j < restClass.methods.length; j++) {
-      var restMethod = restClass.methods[j];
-      if (restMethod.fullName === name) {
-        return restMethod;
+/**
+ * gets or creates the rest method by name cache map.
+ * @param reload forces the map to refresh if true
+ * @returns {boolean} returns true if the cache was updated
+ */
+RestAdapter.prototype.getOrCreateRestMethodByNameMap = function(reload) {
```

Comment:
1. Let's make this method private please - add `_` prefix to the method name.
2. Boolean arguments are difficult to understand - the line `this.getOrCreateRestMethodByNameMap(true);` below does not make the intent (reload the cache) clear.
3. I think this method is doing too much and should be simplified to rebuilding the cache only. Let the caller handle reload.